catalog/models.py

```diff
@@ -1,3 +1,94 @@
 from django.db import models
+from django.urls import reverse  # Used to generate URLs by reversing the URL patterns
+import uuid  # Required for unique book instances
 
-# Create your models here.
+
+class Genre(models.Model):
+    """Model representing a book genre."""
+
+    name = models.CharField(
+        max_length=200, help_text='Enter a book genre (e.g.Science Fiction)'
+    )
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.name
+
+
+class Book(models.Model):
+    """Model representing a book (but not a specific copy of a book)."""
+
+    title = models.CharField(max_length=200)
+    author = models.ForeignKey('Author', on_delete=models.SET_NULL, null=True)
```

Comment:
Tìm hiểu các giá trị khác của option `on_delete`

Reply:
CASCADE: Khi đối tượng cha bị xóa, tất cả các đối tượng con liên quan sẽ bị xóa theo. 

PROTECT: Cản trở việc xóa đối tượng cha nếu có các đối tượng con liên quan tồn tại.

SET_NULL: Thiết lập giá trị của trường ForeignKey trong các đối tượng con thành NULL khi đối tượng cha bị xóa. (yêu cầu trường ForeignKey phải có null=True)

SET_DEFAULT: Thiết lập giá trị của trường ForeignKey trong các đối tượng con thành giá trị mặc định đã được xác định khi đối tượng cha bị xóa.

SET(): Thiết lập giá trị của trường ForeignKey trong các đối tượng con thành giá trị đã chỉ định bởi hàm hoặc giá trị có sẵn khi đối tượng cha bị xóa.

DO_NOTHING: Không thực hiện hành động gì khi đối tượng cha bị xóa. Phụ thuộc vào xử lý hành động sau này.

RESTRICT: Tương tự như PROTECT, được sử dụng để đảm bảo việc xóa đối tượng cha không thể thực hiện nếu nó ảnh hưởng đến tính toàn vẹn dữ liệu trong cơ sở dữ liệu.